Added getters and setters for widgets

samuelznewton opened the pull request:
Per [issue 408](https://github.com/pyglet/pyglet/issues/408), this pull adds a value property to pyglet.gui widgets.  It must be implemented separately for each widget, and it is currently implemented for all existing widgets.  I've done a little bit of testing and it seems to work properly.  This is my first time contributing to an open-source project, so I'm really just hoping I've done everything right.  If not, please let me know, and I'll be glad to try to make things right!

benmoran56 commented on pyglet/gui/widgets.py:
```diff
@@ -73,6 +73,17 @@ def height(self):
+    # can be 'activated' or changed without requiring user input.
+    # The normal events for control activation should NOT be triggered.
+    @property
+    def value(self):
```
Rather than having all of the block comments, we should probably just add in the doc strings. Something like: 
"""Query or set the Widget's value.

This properly allows you to set the value of the Widget directly, 
without any user input. This can be useful to restore the Widgets
to a previous state. Note that events are not dispatched when
changing the value directly. 
"""

benmoran56 commented on pyglet/gui/widgets.py:
```diff
@@ -125,6 +136,17 @@ def __init__(self, x, y, pressed, depressed, hover=None, batch=None, group=None)
+    # Override
+    @value.setter
+    def value(self, value):
+        self._pressed = value
```
It might be a good idea to add assertions in the setters, to prevent strange bugs. 
Something like: 
```py
assert type(value) is bool, "Value must be True or False."
or
assert type(value) in (int, float), "Value must be an int or float."
```

benmoran56 reviewed (changes requested):
Looks good overall! I made some inline comments, and we should probably get rid of the # comments, but other than that you're on the right track. 

I think however, that the PushButton probably won't work well with this, because it's a momentary Widget. Instead, we can move those property/setter down to the ToggleButton, where they would work as expected. We

benmoran56 commented:
Also, congratulations on your first contribution!

samuelznewton commented:
> I think however, that the PushButton probably won't work well with this, because it's a momentary Widget. Instead, we can move those property/setter down to the ToggleButton, where they would work as expected. We

I see your point, but that's only if this is being used for the one use case you have in mind: restoring a previous state.  However, I think there are more uses than that.  I've actually brought up this issue for a different use case, where an event in my program triggers the change of a boolean - a boolean normally controlled by a button.  In my case, yes, it's a ToggleButton.

But imagine that you're trying to program a game where the player has to push some on-screen buttons.  And suppose you wanted to create a tutorial, and you show what happens when you press certain buttons.  Then it makes perfect sense to control a PushButton through code, I would think.

Although, in order to make this more effective, it would probably be a good idea to allow a developer to disable a Widget - perhaps a boolean property.  This, too, would have more uses than just the one I described.  I could actually use a disable feature in my program, as well.

samuelznewton commented:
I added the docstring to the value property, removed all of my # comments, and added assertions.  I did not move the value property definition out of PushButton into ToggleButton, though I can if you still feel that's best.

If you would like a disable feature, I can probably add that, too.  Though, I don't know whether that would be something to add to this pull request or something totally separate, so if you do want that, please let me know how to proceed.

benmoran56 commented:
The changes look good!

I understand your point about the Push Button widget. That does make sense. Lets leave the potential "disable" discussion for a separate PR, so I think we can merge this in for now.
